Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM ruby:2.4.3
+
+WORKDIR /app
+COPY Gemfile ./Gemfile
+COPY Gemfile.lock ./Gemfile.lock
+RUN apt-get update \
```

Comment:
Split the
```
RUN apt-get update \\
    && apt-get install -y mysql-client postgresql-client sqlite3 --no-install-recommends \
    && rm -rf /var/lib/apt/lists/* \
    && gem update --system \
    && gem install bundler -v '1.16.1'
```
Move it before the copy of the gemfile and the lock
This way most of the base image remains the same since those packages tend to change way less often than our gem list.

Reply:
@hugocorbucci This is better?
```
RUN apt-get update \
    && apt-get install -y mysql-client postgresql-client sqlite3 --no-install-recommends \
    && rm -rf /var/lib/apt/lists/* \
    && gem update --system \
    && gem install bundler -v '1.16.1' 

COPY Gemfile ./Gemfile
COPY Gemfile.lock ./Gemfile.lock

RUN bundle install -j 16
```
Do you want to keep the `gem update --system` together with `apt-get` or together with `gem install`?

Reply:
At that point I assume the gems would mostly be core ruby gems like rubygems itself so it should be fine to keep in the same... We could just split that in it's own RUN just in case

Reply:
@hugocorbucci Like a slipt just for `gem update... gem install` or put this together with `bundle install`?

Reply:
Or leave as it is?

Reply:
I think maybe split the `gem update --system && gem install bundler ...` into their own RUN. It's unlikely (I think) that those would change very often but the intermediary layer won't cost much and if they do change, don't have to re-aptget.